integrations/langfuse/example/basic_rag.py

```diff
@@ -0,0 +1,67 @@
+import os
+
+os.environ["LANGFUSE_HOST"] = "https://cloud.langfuse.com"
+os.environ["TOKENIZERS_PARALLELISM"] = "false"
+os.environ["HAYSTACK_CONTENT_TRACING_ENABLED"] = "true"
+
+from datasets import load_dataset
+
+from haystack.document_stores.in_memory import InMemoryDocumentStore
+from haystack import Document, Pipeline
+from haystack.components.embedders import SentenceTransformersDocumentEmbedder, SentenceTransformersTextEmbedder
+from haystack.components.retrievers import InMemoryEmbeddingRetriever
+from haystack.components.generators import OpenAIGenerator
+from haystack.components.builders import PromptBuilder
+
+from haystack_integrations.components.others.langfuse import LangfuseComponent
+
+
+def get_pipeline(document_store: InMemoryDocumentStore):
+    retriever = InMemoryEmbeddingRetriever(document_store=document_store, top_k=2)
+
+    template = """
+    Given the following information, answer the question.
+
+    Context:
+    {% for document in documents %}
+        {{ document.content }}
+    {% endfor %}
+
+    Question: {{question}}
+    Answer:
+    """
+
+    prompt_builder = PromptBuilder(template=template)
+
+    basic_rag_pipeline = Pipeline()
+    # Add components to your pipeline
+    basic_rag_pipeline.add_component("tracer", LangfuseComponent("Basic RAG Pipeline"))
+    basic_rag_pipeline.add_component(
+        "text_embedder", SentenceTransformersTextEmbedder(model="sentence-transformers/all-MiniLM-L6-v2")
+    )
+    basic_rag_pipeline.add_component("retriever", retriever)
+    basic_rag_pipeline.add_component("prompt_builder", prompt_builder)
+    basic_rag_pipeline.add_component("llm", OpenAIGenerator(model="gpt-3.5-turbo", generation_kwargs={"n": 2}))
+
+    # Now, connect the components to each other
+    # NOTE: the tracer component doesn't need to be connected to anything in order to work
+    basic_rag_pipeline.connect("text_embedder.embedding", "retriever.query_embedding")
+    basic_rag_pipeline.connect("retriever", "prompt_builder.documents")
+    basic_rag_pipeline.connect("prompt_builder", "llm")
+
+    return basic_rag_pipeline
+
+
+if __name__ == "__main__":
+    document_store = InMemoryDocumentStore()
+    dataset = load_dataset("bilgeyucel/seven-wonders", split="train")
+    embedder = SentenceTransformersDocumentEmbedder("sentence-transformers/all-MiniLM-L6-v2")
+    embedder.warm_up()
+    docs_with_embeddings = embedder.run([Document(**ds) for ds in dataset]).get("documents") or []  # type: ignore
+    document_store.write_documents(docs_with_embeddings)
+
+    pipeline = get_pipeline(document_store)
+    question = "What does Rhodes Statue look like?"
+    response = pipeline.run({"text_embedder": {"text": question}, "prompt_builder": {"question": question}})
```

Comment:
I actually get an error message when I try to run this code:
```
AttributeError                            Traceback (most recent call last)
[<ipython-input-7-e4c16a4f3735>](https://localhost:8080/#) in <cell line: 10>()
      8 pipeline = get_pipeline(document_store)
      9 question = "What does Rhodes Statue look like?"
---> 10 response = pipeline.run({"text_embedder": {"text": question}, "prompt_builder": {"question": question}})
     11 
     12 print(response["llm"]["replies"][0])

2 frames
[/usr/local/lib/python3.10/dist-packages/haystack_integrations/tracing/langfuse/tracer.py](https://localhost:8080/#) in <listcomp>(.0)
     31         elif key.endswith(".output"):
     32             if "replies" in value:
---> 33                 replies = [m.to_openai_format() for m in value["replies"]]
     34                 self._span.update(output=replies)
     35             else:

AttributeError: 'str' object has no attribute 'to_openai_format'
```

Reply:
Thanks @annthurium , fixed with https://github.com/deepset-ai/haystack-core-integrations/pull/686/commits/e27d12020cc7d7a8aa67707104665aa60d9c524c

Reply:
sorry, but how do I run the code with that patch? still running into the bug. https://colab.research.google.com/drive/1w54Pb1TnpL0svOUB21t4owG7apklcsWf#scrollTo=6b3KXSe2mp6T

Reply:
Aha, in this case you need to install directly from langfuse_integration github branch rather than oficial install pulled from pypi server - the new code is not there yet.